src/main/java/entities/User.java

```diff
@@ -0,0 +1,170 @@
+package java.entities;
+import java.util.ArrayList;
+
+public class User {
+    // ArrayList of User's favourited posts
+    private final ArrayList<Post> favourites;
+
+    // ArrayList of Posts User has made
+    private final ArrayList<Post> posts;
+
+    // boolean relating if User has admin privileges
+    private final boolean isAdmin;
+
+    // Number of Users created
+    public static int numUsers = 0;
+
+    // id is the unique identifier of User, equal to number of posts at time of creation
+    private final int id;
+
+    // password is the user's input password, used to sign in
+    private final String password;
+
+    // email is the users input email, used to sign in
+    private final String email;
+
+    /*
+    * Initializes an instance of User
+    *
+    * @param admin whether the user will have admin priviledges
+    * */
+    public User(boolean admin, int numUsersCreated, String email, String password) {
+        this.favourites = new ArrayList<Post>();
+        this.posts = new ArrayList<Post>();
+        this.isAdmin = admin;
+        User.numUsers = numUsersCreated;
+        User.numUsers ++;
+        this.id = User.numUsers;
+        this.email = email;
+        this.password = password;
+    }
+
+    /*
+    * Returns User's favourites
+    *
+    * @return User's favourites attribute
+    * */
+    public ArrayList getFavourites() {
+        /* Return an ArrayList containing the favourites of this User */
+        return this.favourites;
+    }
+
+    /*
+    * Adds a Post to User's favourites
+    *
+    * @param toAdd The Post to add to User's favourites
+    * */
+    public void addFavourite(Post toAdd) {
+        /* Add Post to this.favourites */
+        this.favourites.add(toAdd);
+    }
+
+    /*
+    * Check if a Post is in User's favourites and remove it if so
+    *
+    * @param toRemove Post to be removed
+    * @return boolean representing success or failure
+    * */
+    public boolean removeFavourite(Post toRemove) {
```

Comment:
Same with comment on line 104, make name more descriptive of what it actually does without conflicting with another use case, etc.